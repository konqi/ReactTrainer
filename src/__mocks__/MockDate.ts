export class MockDate extends Date {
  constructor(datetime?: number) {
    super(datetime || 3141592653589)
    return this
  }

  static now() {
    return 3141592653589
  }
}
