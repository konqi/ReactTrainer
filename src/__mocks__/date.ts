const originalDate = global.Date

export const freezeTime = (clockFrozenAt: number = 3141592653589): Date =>
  //@ts-ignore
  class extends originalDate {
    constructor(datetime?: number | string | Date) {
      super((typeof datetime !== undefined && datetime) || clockFrozenAt)
      return this
    }

    static now() {
      return clockFrozenAt
    }
  }

export const restoreCausality = () => originalDate
