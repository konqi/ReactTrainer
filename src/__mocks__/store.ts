import {createStore} from 'redux'
import {ApplicationState} from '../state'
import {rootReducer} from '../state/rootReducer'
import {Trainee} from '../types/Trainee'
import {Session} from '../types/Session'

export const createMockStore = (initialApplicationState: ApplicationState) =>
  createStore(rootReducer, initialApplicationState)

export class TraineeBuilder {
  trainee: Trainee
  constructor() {
    this.trainee = {
      id: 'traineeId',
      name: 'Dummy Trainee',
      price: 15,
    } as Trainee
  }

  withId(id: string) {
    this.trainee.id = id
    return this
  }

  withName(name: string) {
    this.trainee.name = name
    return this
  }

  withPrice(price: number) {
    this.trainee.price = price
    return this
  }

  withSessionRef(sessionRef: string[]) {
    this.trainee.sessionsRef = sessionRef
    return this
  }

  build() {
    return this.trainee
  }
}

export class SessionBuilder {
  session: Session

  constructor(traineeId: string) {
    this.session = {
      id: 'sessionId',
      datetime: new Date(),
      payedAmount: 15,
      traineeRef: traineeId,
    }
  }

  withId(id: string) {
    this.session.id = id
    return this
  }

  withDatetime(datetime: Date) {
    this.session.datetime = datetime
    return this
  }

  withDescription(description: string) {
    this.session.description = description
    return this
  }

  withPayedAmount(payedAmount: number) {
    this.session.payedAmount = payedAmount
    return this
  }

  withPrice(price: number) {
    this.session.price = price
    return this
  }

  build() {
    return this.session
  }
}
