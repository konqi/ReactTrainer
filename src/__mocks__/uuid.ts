export const v1 = () => "1";
export const v4 = () => "4";
