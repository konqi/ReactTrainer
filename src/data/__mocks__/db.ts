export const deleteMock = jest.fn()
export const docMock = jest.fn()

export const db = {
  batch: () => ({
    delete: deleteMock,
  }),
  collection: () => ({
    doc: docMock,
  }),
}
