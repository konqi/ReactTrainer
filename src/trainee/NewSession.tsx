import {
  Button,
  createStyles,
  Grid,
  makeStyles,
  TextField,
  Theme,
} from '@material-ui/core'
import {Add as AddIcon} from '@material-ui/icons'
import {format, isFuture} from 'date-fns'
import React, {ChangeEvent, FormEvent} from 'react'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    center: {alignContent: 'center'},
  })
)

interface ExternalProps {
  datetime?: Date
  description?: string
  payedAmount?: number
  onDatetimeChange?: (date: Date) => void
  onDescriptionChange?: (description: string) => void
  onPayedAmountChange?: (payedAmount: number) => void
  onSubmit?: () => void
}

export const NewSession: React.FC<ExternalProps> = ({
  datetime = new Date(),
  description = undefined,
  payedAmount = 0,
  onDatetimeChange = () => {},
  onDescriptionChange = () => {},
  onPayedAmountChange = () => {},
  onSubmit = () => {},
}) => {
  const classes = useStyles()

  const handleStringChange = (callback: (newValue: string) => void) => (
    event: ChangeEvent<HTMLInputElement>
  ) => {
    callback(event.target.value)
  }

  const handleNumberChange = (callback: (newValue: number) => void) => (
    event: ChangeEvent<HTMLInputElement>
  ) => {
    callback(Number(event.target.value))
  }

  const handleChangeDate = (event: ChangeEvent<HTMLInputElement>) => {
    const result = /(\d{4})-(\d{2})-(\d{2})/.exec(event.target.value)
    const [, year, month, day] = result as String[]
    const newDate = new Date(datetime.getTime())
    newDate.setDate(Number(day))
    newDate.setMonth(Number(month) - 1)
    newDate.setFullYear(Number(year))
    onDatetimeChange(newDate)
  }

  const handleChangeTime = (event: ChangeEvent<HTMLInputElement>) => {
    const result = /(\d{2}):(\d{2})/.exec(event.target.value)
    const [, hours, minutes] = result as String[]
    const newDate = new Date(datetime.getTime())
    newDate.setHours(Number(hours))
    newDate.setMinutes(Number(minutes))
    newDate.setSeconds(0)
    newDate.setMilliseconds(0)
    onDatetimeChange(newDate)
  }

  const handleSubmit = (event: FormEvent) => {
    event.preventDefault()
    onSubmit()
  }

  const date = format(datetime, 'YYYY-MM-DD')
  const time = format(datetime, 'HH:mm')

  return (
    <form onSubmit={handleSubmit}>
      <Grid container direction="column" spacing={2} className={classes.center}>
        <Grid item>
          <TextField
            id="TrainingDateInput"
            type="date"
            variant="outlined"
            label="Datum"
            defaultValue={date}
            fullWidth
            onChange={handleChangeDate}
          />
        </Grid>
        {/* if date in future, show time */}
        {isFuture(datetime) && (
          <Grid item>
            <TextField
              id="TrainingTimeInput"
              type="time"
              variant="outlined"
              label="Uhrzeit"
              defaultValue={time}
              fullWidth
              onChange={handleChangeTime}
            />
          </Grid>
        )}
        {/* payed? amount */}
        <Grid item>
          <TextField
            id="TrainingPayedAmountInput"
            type="number"
            variant="outlined"
            value={payedAmount}
            label="Bezahlt"
            fullWidth
            onChange={handleNumberChange(onPayedAmountChange)}
          />
        </Grid>
        {/* description */}
        <Grid item>
          <TextField
            id="TrainingDescriptionInput"
            multiline
            variant="outlined"
            label="Beschreibung"
            value={description}
            fullWidth
            onChange={handleStringChange(onDescriptionChange)}
          />
        </Grid>

        <Grid item>
          <Button fullWidth type="submit" variant="contained" color="primary">
            Hinzuf√ºgen
            <AddIcon />
          </Button>
        </Grid>
      </Grid>
    </form>
  )
}
