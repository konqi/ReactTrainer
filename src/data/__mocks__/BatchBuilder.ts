import {Collection} from '../../types/Collection'
import {db} from '../db'

const batchDeleteMock = jest.fn()
const batchCommitMock = jest.fn()

const BatchBuilderMock = {
  delete: batchDeleteMock,
  execute: batchCommitMock,
}

export class BatchBuilder {
  constructor() {
    return BatchBuilderMock
  }
}
