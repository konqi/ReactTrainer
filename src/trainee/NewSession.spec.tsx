import {fireEvent, render, cleanup} from '@testing-library/react'
import * as React from 'react'
import {NewSession} from './NewSession'
import {freezeTime, restoreCausality} from '../__mocks__/date'

describe('snapshot tests', () => {
  beforeAll(() => {
    // @ts-ignore
    global.Date = freezeTime(3141592653589)
  })

  afterAll(() => {
    global.Date = restoreCausality()
  })

  afterEach(cleanup)

  it('should render without parameters', () => {
    const {baseElement, unmount} = render(<NewSession />)
    expect(baseElement).toMatchSnapshot()
    unmount()
  })

  it('should render with parameters', () => {
    const datetime = new Date()
    const description = ''
    const payedAmount = 10
    const {baseElement, unmount} = render(
      <NewSession {...{datetime, description, payedAmount}} />
    )
    expect(baseElement).toMatchSnapshot()
    unmount()
  })

  it('should render time select for future dates', () => {
    const datetime = new Date(Date.now() + 24 * 60 ** 2 * 1000) // tomorrow
    const description = ''
    const payedAmount = 10
    const {baseElement, unmount} = render(
      <NewSession {...{datetime, description, payedAmount}} />
    )
    expect(baseElement).toMatchSnapshot()
    unmount()
  })
})

describe('integration tests', () => {
  beforeAll(() => {
    // @ts-ignore
    global.Date = freezeTime(3141592653589)
  })

  afterAll(() => {
    global.Date = restoreCausality()
  })

  afterEach(cleanup)

  it('should update datetime according to date input change', () => {
    const mockFn = jest.fn()
    const {getByLabelText, unmount} = render(
      <NewSession onChange={mockFn} datetime={new Date(0)} />
    )

    expect(mockFn).not.toHaveBeenCalled()
    fireEvent.change(getByLabelText('Datum'), {target: {value: '3333-11-22'}})
    const expectedDateTime = new Date(0)
    expectedDateTime.setFullYear(3333)
    expectedDateTime.setMonth(10)
    expectedDateTime.setDate(22)
    expect(mockFn).toHaveBeenCalledWith('date', expectedDateTime)

    unmount()
  })

  it('should update datetime according to time input change', () => {
    const mockFn = jest.fn()

    const future = new Date(Date.now() + 24 * 60 ** 2 * 1000) // tomorrow
    const {getByLabelText, unmount} = render(
      <NewSession datetime={future} onChange={mockFn} />
    )

    expect(mockFn).not.toHaveBeenCalled()
    fireEvent.change(getByLabelText('Uhrzeit'), {target: {value: '12:21'}})
    const expectedDateTime = future
    expectedDateTime.setHours(12)
    expectedDateTime.setMinutes(21)
    expectedDateTime.setSeconds(0)
    expectedDateTime.setMilliseconds(0)
    expect(mockFn).toHaveBeenCalledWith('time', expectedDateTime)

    unmount()
  })

  it('should update description', () => {
    const mockFn = jest.fn()
    const {getByLabelText, getByText, unmount} = render(
      <NewSession onChange={mockFn} />
    )

    expect(mockFn).not.toHaveBeenCalled()
    fireEvent.change(getByLabelText('Beschreibung'), {
      target: {value: 'Hello World!'},
    })
    expect(mockFn).toHaveBeenCalledWith('description', 'Hello World!')

    unmount()
  })

  it('should update payed amount', () => {
    const mockFn = jest.fn()
    const {getByLabelText, getByText, unmount} = render(
      <NewSession onChange={mockFn} />
    )

    expect(mockFn).not.toHaveBeenCalled()
    fireEvent.change(getByLabelText('Bezahlt'), {target: {value: '10'}})
    expect(mockFn).toHaveBeenCalledWith('payedAmount', 10)

    unmount()
  })

  it('should fire submit callback', () => {
    const mockFn = jest.fn()
    const {getByLabelText, getByText, getByRole, unmount} = render(
      <NewSession onSubmit={mockFn} />
    )

    expect(mockFn).not.toHaveBeenCalled()
    fireEvent.submit(getByRole('form'))
    expect(mockFn).toHaveBeenCalled()

    unmount()
  })
})
