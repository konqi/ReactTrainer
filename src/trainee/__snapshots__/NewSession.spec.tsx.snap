// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests should render time select for future dates 1`] = `
<body>
  <div />
  <div />
  <div>
    <form>
      <div
        class="MuiGrid-root makeStyles-center-403 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="SessionDateInput"
            >
              Datum
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-563 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-564"
                  style="width: 0px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="SessionDateInput"
                type="date"
                value="2069-07-22"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="SessionTimeInput"
            >
              Uhrzeit
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-563 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-564"
                  style="width: 0px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="SessionTimeInput"
                type="time"
                value="01:37"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="SessionPayedAmountInput"
            >
              Bezahlt
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-563 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-564"
                  style="width: 0px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="SessionPayedAmountInput"
                type="number"
                value="10"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="SessionDescriptionInput"
            >
              Beschreibung
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-563 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-564"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                id="SessionDescriptionInput"
                rows="1"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; width: 100%;"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Hinzufügen
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</body>
`;

exports[`snapshot tests should render with parameters 1`] = `
<body>
  <div />
  <div>
    <form>
      <div
        class="MuiGrid-root makeStyles-center-202 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="SessionDateInput"
            >
              Datum
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-362 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-363"
                  style="width: 0px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="SessionDateInput"
                type="date"
                value="2069-07-21"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="SessionPayedAmountInput"
            >
              Bezahlt
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-362 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-363"
                  style="width: 0px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="SessionPayedAmountInput"
                type="number"
                value="10"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="SessionDescriptionInput"
            >
              Beschreibung
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-362 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-363"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                id="SessionDescriptionInput"
                rows="1"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; width: 100%;"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Hinzufügen
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</body>
`;

exports[`snapshot tests should render without parameters 1`] = `
<body>
  <div>
    <form>
      <div
        class="MuiGrid-root makeStyles-center-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="SessionDateInput"
            >
              Datum
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-161 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-162"
                  style="width: 0px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="SessionDateInput"
                type="date"
                value="2069-07-21"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="SessionPayedAmountInput"
            >
              Bezahlt
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-161 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-162"
                  style="width: 0px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="SessionPayedAmountInput"
                type="number"
                value="0"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="SessionDescriptionInput"
            >
              Beschreibung
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-161 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-162"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                id="SessionDescriptionInput"
                rows="1"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; width: 100%;"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Hinzufügen
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</body>
`;
